language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/codeaction/CodeActionNodeValidatorTest.java

```diff
@@ -0,0 +1,64 @@
+package org.ballerinalang.langserver.codeaction;
+
+import io.ballerina.compiler.syntax.tree.NonTerminalNode;
+import io.ballerina.compiler.syntax.tree.SyntaxTree;
+import io.ballerina.tools.text.TextDocument;
+import io.ballerina.tools.text.TextDocuments;
+import org.ballerinalang.langserver.common.utils.CommonUtil;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * Test cases to validate the code action syntax.
+ *
+ * @since 2.0.0
+ */
+public class CodeActionNodeValidatorTest {
+
+    private final Path sourcesPath = new File(getClass().getClassLoader().getResource("codeaction/node-validator")
+            .getFile()).toPath();
```

Comment:
`getResource()` is annotated `nullable`, shall we prevent `getFile()` method invocation producing NPE?